net-p2p/retroshare/retroshare-0.6.6.ebuild

```
@@ -0,0 +1,124 @@
+# Copyright 1999-2021 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+
+inherit desktop edos2unix qmake-utils xdg
+
+DESCRIPTION="P2P private sharing application"
+HOMEPAGE="https://retroshare.cc"
+SRC_URI="http://download.opensuse.org/repositories/network:/retroshare/Debian_Testing/retroshare-common_${PV}.orig.tar.gz -> ${P}.tar.gz"
+S="${WORKDIR}/RetroShare"
+
+LICENSE="AGPL-3 Apache-2.0 CC-BY-SA-4.0 GPL-2 GPL-3 LGPL-3"
+SLOT="0"
+KEYWORDS="~amd64 ~x86"
+IUSE="autologin cli +gui +jsonapi libupnp +miniupnp +service +sqlcipher"
+
+REQUIRED_USE="
+	|| ( gui service )
+	?? ( libupnp miniupnp )
+	service? ( || ( cli jsonapi ) )"
+
+RDEPEND="
+	app-arch/bzip2
+	dev-libs/openssl:0=
+	>=dev-libs/rapidjson-1.1.0
+	sys-libs/zlib
+	autologin? ( app-crypt/libsecret )
+	gui? (
+		dev-qt/qtcore:5
+		dev-qt/qtmultimedia:5
+		dev-qt/qtnetwork:5
+		dev-qt/qtprintsupport:5
+		dev-qt/qtscript:5
+		dev-qt/qtxml:5
+		dev-qt/qtgui:5
+		dev-qt/qtwidgets:5
+		dev-qt/qtx11extras:5
+		x11-libs/libX11
+		x11-libs/libXScrnSaver
+	)
+	libupnp? ( net-libs/libupnp:= )
+	miniupnp? ( net-libs/miniupnpc:= )
+	service? ( dev-qt/qtcore:5 )
+	sqlcipher? ( dev-db/sqlcipher )
+	!sqlcipher? ( dev-db/sqlite:3 )"
+
+DEPEND="${RDEPEND}
+	dev-qt/qtcore:5
+	gui? ( dev-qt/designer:5 )"
+
+BDEPEND="dev-util/cmake
+	virtual/pkgconfig
+	jsonapi? ( app-doc/doxygen )"
+
+src_prepare() {
+	# CRLF endings break patch...
+	edos2unix retroshare-gui/src/gui/elastic/elnode.h
+	default
+}
+
+src_configure() {
+	local qconfigs=(
+		$(usex cli '' 'no_')rs_service_terminal_login
+		$(usex autologin '' 'no_')rs_autologin
+		$(usex gui '' 'no_')retroshare_gui
+		$(usex jsonapi '' 'no_')rs_jsonapi
+		$(usex service '' 'no_')retroshare_service
+		$(usex sqlcipher '' 'no_')sqlcipher
+	)
+
+	local qupnplibs="none"
+	use miniupnp && qupnplibs="miniupnpc"
+	use libupnp && qupnplibs="upnp ixml"
+
+	eqmake5 CONFIG+="${qconfigs[*]}" \
+		RS_MAJOR_VERSION=$(ver_cut 1) RS_MINOR_VERSION=$(ver_cut 2) \
+		RS_MINI_VERSION=$(ver_cut 3) RS_EXTRA_VERSION="-gentoo-${PR}" \
+		RS_UPNP_LIB="${qupnplibs}"
+}
+
+src_install() {
+	use gui && dobin retroshare-gui/src/retroshare
+	use service && dobin retroshare-service/src/retroshare-service
+
+	insinto /usr/share/retroshare
+	doins libbitdht/src/bitdht/bdboot.txt
+
+	dodoc README.asciidoc
+
+	if use gui; then
+		make_desktop_entry retroshare
+
+		for i in 24 48 64 128 ; do
+			doicon -s ${i} "data/${i}x${i}/apps/retroshare.png"
+		done
+	fi
+}
+
+pkg_preinst() {
```

No longer need to define the postinst/postrm with xdg.eclass, but it has a preinst and being overwritten here.
```suggestion
pkg_preinst() {
	xdg_pkg_preinst

```

Or else it leads to:
```
 * QA Notice: new icons were found installed but icon cache
 * has not been updated:
 *   /usr/share/icons/hicolor/64x64/apps/retroshare.png
```